components/org.wso2.carbon.identity.application.authenticator.iwa/src/main/java/org/wso2/carbon/identity/application/authenticator/iwa/IWAFederatedAuthenticator.java

```diff
@@ -158,4 +188,90 @@ private String getAuthenticatedUserFromToken(GSSCredential gssCredentials, byte[
         }
     }
 
+    private boolean IsUserExistsInListedUserStores(String authenticatedUserName, String tenantDomain, String
+            userStoreDomains) throws AuthenticationFailedException {
+        boolean isUserExists = false;
+        try {
+            for (String userStoreDomain : userStoreDomains.split(",")
+                    ) {
+                if (isExistsInUserStore(authenticatedUserName, tenantDomain, userStoreDomain)) {
+                    isUserExists = true;
+                }
+            }
+            return isUserExists;
+        } catch (AuthenticationFailedException e) {
+            throw new AuthenticationFailedException("IWAFederatedAuthenticator " +
```

Comment:
In this case the authenticator will act as a Local Authenticator, Shall we change the logs accordingly or simply use IWAApplicationAuthenticator in logs?